package/MDAnalysis/analysis/waterdynamics.py

```diff
@@ -820,7 +825,8 @@ def _selection_serial(self, universe, selection_str):
         selection = []
         for ts in universe.trajectory:
             selection.append(universe.select_atoms(selection_str))
-            print(ts.frame)
+            sys.stdout.write("\rSelecting atoms from frame: %i"%ts.frame)
```

Comment:
Maybe we misunderstand each other (?): I am not advocating to use a logger but use the `ProgressMeter` class which properly encapsulates the use case here: write status for every iteration in a loop. Under the hood it uses `sys.stderr` and flushes but with added goodies like #928 / #960 (and some other neat things that @jbarnoud has shown or promised (IIRC... ;-) ) such as real progress bars, including nice ones in Jupyter notebooks) it makes sense to unify the way we show progress.